### PR TITLE
Add support for extra parameters kubelet configuration

### LINKED ISSUE #795: Feature request: option to configure kubelet to reserve resources for system daemons
**Why do you want this feature?**
In EKS, nodes that starve for resources for system daemons go to *NotReady* state and stay there until someone manually deletes the node in EC2 UI. The feature of a timeout from the master to a node if it stays *NotReady* for too long because it can't comunicate will come with kubernetes 1.15 (in 2020). Reserving resources is very important [https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/](url)

**What feature/behavior/change do you want?**
There should be a field in the cluster config yaml to configure this settings and that applies them to the userdata in the cloudformation template, something like `extraKubeletFlags` or `appendToKubeletConfig`

it would look like:

```
appendToKubeletConfig:
    kubeReserved:
      cpu: "300m"
      memory: "300Mi"
      ephemeral-storage: "1Gi"
    kubeReservedCgroup: "/kube-reserved"
    systemReserved:
      cpu: "300m"
      memory: "300Mi"
      ephemeral-storage: "1Gi"
    evictionHard:
      memory.available:  "200Mi"
      nodefs.available: "5%"
```

### PR DESCRIPTION
**Pull request by martina-if**
closes #795

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] Manually tested
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 1]
```diff
@@ -221,3 +225,27 @@ func countEnabledFields(fields ...*string) int {
 	}
 	return count
 }
+
+func validateNodeGroupKubeletExtraConfig(kubeletConfig *NodeGroupKubeletConfig) error {
+	if kubeletConfig == nil {
+		return nil
+	}
+
+	var kubeletForbiddenFields = map[string]bool{
```
Comment (rndstr) [inline pkg/apis/eksctl.io/v1alpha5/validation.go]: nit: `map[string]struct{}` is usually used in places where the value doesn't matter

**Comment by errordeveloper** [comment]
@martina-if are you mostly happy here, or do you want to add anything, e.g. an example?

**Comment by martina-if** [comment]
I want to add some documentation

File: pkg/apis/eksctl.io/v1alpha5/validation.go (hunk 1 above)
Comment (martina-if) [reply pkg/apis/eksctl.io/v1alpha5/validation.go]: thanks!

File: pkg/apis/eksctl.io/v1alpha5/validation_test.go [hunk 2]
```diff
@@ -5,129 +5,169 @@ import (
...
+			BeforeEach(func() {
+				ng = &NodeGroup{}
+			})
+
+			It("Forbids overriding basic fields", func() {
+				testKeys := []string{"kind", "apiVersion", "address", "clusterDomain", "authentication",
+					"authorization", "serverTLSBootstrap", "featureGates"}
+
+				for _, key := range testKeys {
+					ng.KubeletExtraConfig = &NodeGroupKubeletConfig{
```
Comment (errordeveloper) [inline pkg/apis/eksctl.io/v1alpha5/validation_test.go]: Nice! For some reason I thought you always have to use `map[string]insterface{}` to construct the thing...

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 3]
```diff
@@ -224,3 +228,27 @@ func countEnabledFields(fields ...*string) int {
...
+		"clusterDomain":      {},
+		"authentication":     {},
+		"authorization":      {},
+		"serverTLSBootstrap": {},
+		"featureGates":       {},
+	}
+
+	for k := range *kubeletConfig {
+		if _, exists := kubeletForbiddenFields[k]; exists {
+			return fmt.Errorf("kubelet key %q cannot be overwritten", k)
```
Comment (errordeveloper) [inline pkg/apis/eksctl.io/v1alpha5/validation.go]: how about `"cannot override %q in kubelete config, as it's critical to eksctl functionality"`?

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 4]
```diff
@@ -224,3 +228,27 @@ func countEnabledFields(fields ...*string) int {
...
+
+	var kubeletForbiddenFields = map[string]struct{}{
+		"kind":               {},
+		"apiVersion":         {},
+		"address":            {},
+		"clusterDomain":      {},
+		"authentication":     {},
+		"authorization":      {},
+		"serverTLSBootstrap": {},
+		"featureGates":       {},
```
Comment (errordeveloper) [inline pkg/apis/eksctl.io/v1alpha5/validation.go]: Actually, we may wish to allow appending feature gates, but let's wait until use-cases emerge :)

**Comment by Jeffwan** [comment]
Great!

**Pull request merged**